src/components/Question/Question.js

```diff
@@ -0,0 +1,87 @@
+import { html } from '../../shared';
+import styles from './Question.module.css';
+
+/**
+ * @typedef {Object} QuestionInstance
+ * @property {HTMLElement} question
+ * @property {() => boolean} isAnswerCorrect Function that returns whether the user has correctly answered a question
+ */
+
+/**
+ * @param {(import('../../data/questions').CharacterQuestion|import('../../data/questions').EpisodeOrLocationQuestion)} questionData
+ * @returns {QuestionInstance}
+ */
+function Question(questionData) {
+  const { category } = questionData;
+
+  const questionText = {
+    character: 'Co to za postać?',
+    episode: `Kto grał w odcinku ${questionData.name}?`,
+    // @ts-ignore
+    location: `Które postaci mieszkają w miejscu ${questionData.name}?`,
+  };
+
+  const answerLetter = {
+    0: 'A',
+    1: 'B',
+    2: 'C',
+    3: 'D',
+  };
+
+  const answers =
+    category === 'character'
+      ? questionData.answers.map(
+          (answer, index) => html`<div id="${index}" class="${styles.answer}">
+            <div class="${styles.answerItem}">${answerLetter[index]}</div>
+            <p class="${styles.answerText}">${answer.name}</p>
+          </div>`,
+        )
+      : questionData.answers.map(
+          (answer, index) => html`<div id="${index}" class="${styles.answer}">
+            <div class="${styles.answerItem}"><img src="${answer.image}" alt="${answer.name}" /></div>
+            <p class="${styles.answerText}">${answer.name}</p>
+          </div>`,
+        );
+
+  const question =
+    category === 'character'
+      ? html`<div class="${styles.questionContainer}">
+          <p class="${styles.questionText}">${questionText[category]}</p>
+          <div class="${styles.content}">
+            <div class="${styles.characterImageDiv}">
+              <img
+                src="${
+                  // @ts-ignore
```

Comment:
Ten `@ts-ignore` przestanie być potrzebny jeżeli zrobi się te zmiany w `questions.js` o których pisałem we wcześniejszym komentarzu. TS będzie miał gwarancję, że `.image` istnieje, bo przez ternary operator z góry wie, że mamy obiekt typu `CharacterQuestion`.

Reply:
Super, dziękuję bardzo za wszystkie rady dotyczące TSa. Zaraz postaram się z tym podziałać, chociaż zmiany w questions.js powodują znów, że TS mi krzyczy w innym miejscu.

Reply:
Jakbyś miała problem z tym krzyczeniem to daj znać tu/na Discordzie.